File: convert_decimal64.r
alnum <- '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJELMNOPQRSTUVWXYZ'

encode_decimal64 <- function(num, chars = alnum) {
    chars_list <- unlist(strsplit(chars, split = ""))

    str <- ''
    while(num != 0) {
        str = gsub(" ", "", paste(chars_list[num %% nchar(chars) + 1], str), fixed = TRUE)
        num = num - num %% nchar(chars)
        num = num / nchar(chars)
    }

    str
}

decode_decimal64 <- function(str, chars = alnum) {
    chars_list <- unlist(strsplit(chars, split = ""))
    str_list <- unlist(strsplit(str, split = ""))

    num <- 0
    for (char in str_list) {
        num = num * nchar(chars)
        num = num + which(chars_list == char) - 1
    }

    num
}


num <- 12345
str <- encode_decimal64(num)

print('# Encrypt to decimal62')
print(str)

res <- decode_decimal64(str)

print('# Decrypt to decimal10')
print(res)

File: convert_decimal64.py
alnum = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJELMNOPQRSTUVWXYZ"

def encode_decimal64(num, chars = alnum):
  str = ""
  while num != 0:
    str = chars[num % len(chars)] + str
    num = num - num % len(chars)
    num = num / len(chars)
  return str

def decode_decimal64(string, chars = alnum):
  num = 0
  for char in string:
    num = num * len(chars)
    num = num + chars.index(char)
  return num


num = 12345
string = encode_decimal64(num)

print('# Encrypt to decimal62')
print(string)

print('# Decrypt to decimal10')
print(decode_decimal64(string))


